### PR TITLE
JAMES-3224 Parameterize ConsistencyLevel.{QUORUM, SERIAL}

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/configuration/CassandraConfiguration.java [hunk 1]
```diff
@@ -44,7 +46,20 @@
...
+        "ONE",
+        "TWO",
+        "THREE",
+        "QUORUM",
+        "ALL",
+        "LOCAL_QUORUM",
+        "EACH_QUORUM",
+        "SERIAL",
+        "LOCAL_SERIAL",
+        "LOCAL_ONE");
```
Comment (chibenwa) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/configuration/CassandraConfiguration.java:62]: Shouldn't we use an enum?

File: dockerfiles/run/guice/cassandra/destination/conf/cassandra.properties [hunk 2]
```diff
@@ -33,3 +33,8 @@ cassandra.retryConnection.minDelay=5000
 # chunk.size.message.read=100
 # chunk.size.expunge=100
 # mailbox.blob.part.size=102400
+
+## Consistency Consistency levels used for common queriies
```
Comment (chibenwa) [inline dockerfiles/run/guice/cassandra/destination/conf/cassandra.properties:37]: s/queriies/queries/g

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraACLMapper.java [hunk 3]
```diff
@@ -66,11 +67,14 @@
     private final PreparedStatement conditionalUpdateStatement;
     private final PreparedStatement readStatement;
     private final PreparedStatement deleteStatement;
+    private final ConsistencyLevel consistencyLevel;
 
     @Inject
-    public CassandraACLMapper(Session session, CassandraUserMailboxRightsDAO userMailboxRightsDAO, CassandraConfiguration cassandraConfiguration) {
+    public CassandraACLMapper(Session session, CassandraUserMailboxRightsDAO userMailboxRightsDAO,
+                              CassandraConfiguration cassandraConfiguration, CassandraConsistenciesConfiguration consistenciesConfiguration) {
```
Comment (chibenwa) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraACLMapper.java:74]: Why isn't CassandraConsistenciesConfiguration part of CassandraConfiguration ? It sounds weird to pass two configuration objects...

File: src/site/xdoc/server/config-cassandra.xml [hunk 4]
```diff
@@ -136,6 +136,11 @@
         <dd>Optional. Defaults to 50.<br/> Controls the number of messages to be expunged in parallel.</dd>
         <dt><strong>mailbox.blob.part.size</strong></dt>
         <dd>Optional. Defaults to 102400 (100KB).<br/> Controls the size of blob parts used to store messages.</dd>
+
+        <dt><strong>cassandra.consistency_level.regular</strong></dt>
+        <dd>Optional. Defaults to QUORUM.<br/> <a href="https://docs.datastax.com/en/cassandra-oss/3.x/cassandra/dml/dmlConfigConsistency.html">See possible values</a>.</dd>
```
Comment (chibenwa) [inline src/site/xdoc/server/config-cassandra.xml:141]: What is this value applied to? We miss a `Allows specifying the driver default consistency level.`

File: dockerfiles/run/guice/cassandra-ldap/destination/conf/cassandra.properties [hunk 5]
```diff
@@ -23,3 +23,8 @@ cassandra.retryConnection.minDelay=5000
 # chunk.size.message.read=100
 # chunk.size.expunge=100
 # mailbox.blob.part.size=102400
+
+## Consistency Consistency levels used for common queriies
```
Comment (Arsnael) [inline dockerfiles/run/guice/cassandra-ldap/destination/conf/cassandra.properties:27]: You repeat `Consistency` twice (same comments for similar files)

**Comment by Arsnael** [comment]
```
[951224b1be89709f797e61ba9c91a996ae95a9a3] [ERROR] /james-project/event-sourcing/event-store-cassandra/src/test/scala/org/apache/james/eventsourcing/eventstore/cassandra/CassandraEventStoreExtension.scala:46: error: not enough arguments for constructor EventStoreDao: (session: com.datastax.driver.core.Session, jsonEventSerializer: org.apache.james.eventsourcing.eventstore.cassandra.JsonEventSerializer, consistenciesConfiguration: org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration)org.apache.james.eventsourcing.eventstore.cassandra.EventStoreDao.
[951224b1be89709f797e61ba9c91a996ae95a9a3] [ERROR] Unspecified value parameter consistenciesConfiguration.
[951224b1be89709f797e61ba9c91a996ae95a9a3] [ERROR]     Some(new EventStoreDao(cassandra.getCassandraCluster.getConf, eventSerializer))
[951224b1be89709f797e61ba9c91a996ae95a9a3] [ERROR]          ^
```

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/configuration/CassandraConfiguration.java (hunk 1 above)
Comment (blackheaven) [reply backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/configuration/CassandraConfiguration.java:62]: `ConsistencyLevels` is one, but I do not know how to safely search for `String`

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraACLMapper.java (hunk 3 above)
Comment (blackheaven) [reply mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraACLMapper.java:74]: `CassandraConfiguration` does not parse, it holds a bunch of `int`, checking they are positive.

Here we want plain `ConsistencyLevel`s.

Moreover, it respect the ISP, which is always nice.

**Comment by chibenwa** [comment]
test this please (build rescheduled)

**Comment by chibenwa** [comment]
```
[951224b1be89709f797e61ba9c91a996ae95a9a3] [ERROR] Failed to execute goal net.alchim31.maven:scala-maven-plugin:3.4.4:testCompile (scala-test-compile) on project event-sourcing-event-store-cassandra: wrap: org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
```

**Comment by blackheaven** [comment]
rebased to track a phantom error

**Comment by chibenwa** [comment]
```
[a95ce0d211788da4b137109e03bbb3533782078d] 1) Could not find a suitable constructor in org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration. Classes must have either one (and only one) constructor annotated with @Inject or a zero-argument constructor that is not private.
[a95ce0d211788da4b137109e03bbb3533782078d]   at org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration.class(CassandraConsistenciesConfiguration.java:29)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration
[a95ce0d211788da4b137109e03bbb3533782078d]     for the 3rd parameter of org.apache.james.mailbox.cassandra.mail.CassandraUidProvider.<init>(CassandraUidProvider.java:68)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.mailbox.cassandra.mail.CassandraUidProvider
[a95ce0d211788da4b137109e03bbb3533782078d]     for the 1st parameter of org.apache.james.mailbox.cassandra.CassandraMailboxSessionMapperFactory.<init>(CassandraMailboxSessionMapperFactory.java:108)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.mailbox.cassandra.CassandraMailboxSessionMapperFactory
[a95ce0d211788da4b137109e03bbb3533782078d] 
[a95ce0d211788da4b137109e03bbb3533782078d] 2) Could not find a suitable constructor in org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration. Classes must have either one (and only one) constructor annotated with @Inject or a zero-argument constructor that is not private.
[a95ce0d211788da4b137109e03bbb3533782078d]   at org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration.class(CassandraConsistenciesConfiguration.java:29)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration
[a95ce0d211788da4b137109e03bbb3533782078d]     for the 3rd parameter of org.apache.james.mailbox.cassandra.mail.CassandraModSeqProvider.<init>(CassandraModSeqProvider.java:95)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.mailbox.cassandra.mail.CassandraModSeqProvider
[a95ce0d211788da4b137109e03bbb3533782078d]     for the 2nd parameter of org.apache.james.mailbox.cassandra.CassandraMailboxSessionMapperFactory.<init>(CassandraMailboxSessionMapperFactory.java:108)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.mailbox.cassandra.CassandraMailboxSessionMapperFactory
[a95ce0d211788da4b137109e03bbb3533782078d] 
[a95ce0d211788da4b137109e03bbb3533782078d] 3) Could not find a suitable constructor in org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration. Classes must have either one (and only one) constructor annotated with @Inject or a zero-argument constructor that is not private.
[a95ce0d211788da4b137109e03bbb3533782078d]   at org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration.class(CassandraConsistenciesConfiguration.java:29)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration
[a95ce0d211788da4b137109e03bbb3533782078d]     for the 6th parameter of org.apache.james.mailbox.cassandra.mail.CassandraMessageDAO.<init>(CassandraMessageDAO.java:114)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.mailbox.cassandra.mail.CassandraMessageDAO
[a95ce0d211788da4b137109e03bbb3533782078d]     for the 4th parameter of org.apache.james.mailbox.cassandra.CassandraMailboxSessionMapperFactory.<init>(CassandraMailboxSessionMapperFactory.java:108)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.mailbox.cassandra.CassandraMailboxSessionMapperFactory
[a95ce0d211788da4b137109e03bbb3533782078d] 
[a95ce0d211788da4b137109e03bbb3533782078d] 4) Could not find a suitable constructor in org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration. Classes must have either one (and only one) constructor annotated with @Inject or a zero-argument constructor that is not private.
[a95ce0d211788da4b137109e03bbb3533782078d]   at org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration.class(CassandraConsistenciesConfiguration.java:29)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration
[a95ce0d211788da4b137109e03bbb3533782078d]     for the 2nd parameter of org.apache.james.mailbox.cassandra.mail.CassandraMessageIdToImapUidDAO.<init>(CassandraMessageIdToImapUidDAO.java:86)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.mailbox.cassandra.mail.CassandraMessageIdToImapUidDAO
[a95ce0d211788da4b137109e03bbb3533782078d]     for the 6th parameter of org.apache.james.mailbox.cassandra.CassandraMailboxSessionMapperFactory.<init>(CassandraMailboxSessionMapperFactory.java:108)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.mailbox.cassandra.CassandraMailboxSessionMapperFactory
[a95ce0d211788da4b137109e03bbb3533782078d] 
[a95ce0d211788da4b137109e03bbb3533782078d] 5) Could not find a suitable constructor in org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration. Classes must have either one (and only one) constructor annotated with @Inject or a zero-argument constructor that is not private.
[a95ce0d211788da4b137109e03bbb3533782078d]   at org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration.class(CassandraConsistenciesConfiguration.java:29)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration
[a95ce0d211788da4b137109e03bbb3533782078d]     for the 3rd parameter of org.apache.james.mailbox.cassandra.mail.CassandraMailboxDAO.<init>(CassandraMailboxDAO.java:75)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.mailbox.cassandra.mail.CassandraMailboxDAO
[a95ce0d211788da4b137109e03bbb3533782078d]     for the 9th parameter of org.apache.james.mailbox.cassandra.CassandraMailboxSessionMapperFactory.<init>(CassandraMailboxSessionMapperFactory.java:108)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.mailbox.cassandra.CassandraMailboxSessionMapperFactory
[a95ce0d211788da4b137109e03bbb3533782078d] 
[a95ce0d211788da4b137109e03bbb3533782078d] 6) Could not find a suitable constructor in org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration. Classes must have either one (and only one) constructor annotated with @Inject or a zero-argument constructor that is not private.
[a95ce0d211788da4b137109e03bbb3533782078d]   at org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration.class(CassandraConsistenciesConfiguration.java:29)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration
[a95ce0d211788da4b137109e03bbb3533782078d]     for the 4th parameter of org.apache.james.mailbox.cassandra.mail.CassandraMailboxPathDAOImpl.<init>(CassandraMailboxPathDAOImpl.java:77)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.mailbox.cassandra.mail.CassandraMailboxPathDAOImpl
[a95ce0d211788da4b137109e03bbb3533782078d]     for the 10th parameter of org.apache.james.mailbox.cassandra.CassandraMailboxSessionMapperFactory.<init>(CassandraMailboxSessionMapperFactory.java:108)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.mailbox.cassandra.CassandraMailboxSessionMapperFactory
[a95ce0d211788da4b137109e03bbb3533782078d] 
[a95ce0d211788da4b137109e03bbb3533782078d] 7) Could not find a suitable constructor in org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration. Classes must have either one (and only one) constructor annotated with @Inject or a zero-argument constructor that is not private.
[a95ce0d211788da4b137109e03bbb3533782078d]   at org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration.class(CassandraConsistenciesConfiguration.java:29)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration
[a95ce0d211788da4b137109e03bbb3533782078d]     for the 3rd parameter of org.apache.james.mailbox.cassandra.mail.CassandraMailboxPathV2DAO.<init>(CassandraMailboxPathV2DAO.java:67)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.mailbox.cassandra.mail.CassandraMailboxPathV2DAO
[a95ce0d211788da4b137109e03bbb3533782078d]     for the 11th parameter of org.apache.james.mailbox.cassandra.CassandraMailboxSessionMapperFactory.<init>(CassandraMailboxSessionMapperFactory.java:108)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.mailbox.cassandra.CassandraMailboxSessionMapperFactory
[a95ce0d211788da4b137109e03bbb3533782078d] 
[a95ce0d211788da4b137109e03bbb3533782078d] 8) Could not find a suitable constructor in org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration. Classes must have either one (and only one) constructor annotated with @Inject or a zero-argument constructor that is not private.
[a95ce0d211788da4b137109e03bbb3533782078d]   at org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration.class(CassandraConsistenciesConfiguration.java:29)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration
[a95ce0d211788da4b137109e03bbb3533782078d]     for the 3rd parameter of org.apache.james.mailbox.cassandra.mail.CassandraAttachmentDAOV2.<init>(CassandraAttachmentDAOV2.java:136)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.mailbox.cassandra.mail.CassandraAttachmentDAOV2
[a95ce0d211788da4b137109e03bbb3533782078d]     for the 14th parameter of org.apache.james.mailbox.cassandra.CassandraMailboxSessionMapperFactory.<init>(CassandraMailboxSessionMapperFactory.java:108)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.mailbox.cassandra.CassandraMailboxSessionMapperFactory
[a95ce0d211788da4b137109e03bbb3533782078d] 
[a95ce0d211788da4b137109e03bbb3533782078d] 9) Could not find a suitable constructor in org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration. Classes must have either one (and only one) constructor annotated with @Inject or a zero-argument constructor that is not private.
[a95ce0d211788da4b137109e03bbb3533782078d]   at org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration.class(CassandraConsistenciesConfiguration.java:29)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration
[a95ce0d211788da4b137109e03bbb3533782078d]     for the 4th parameter of org.apache.james.mailbox.cassandra.mail.CassandraACLMapper.<init>(CassandraACLMapper.java:74)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.mailbox.cassandra.mail.CassandraACLMapper
[a95ce0d211788da4b137109e03bbb3533782078d]     for the 19th parameter of org.apache.james.mailbox.cassandra.CassandraMailboxSessionMapperFactory.<init>(CassandraMailboxSessionMapperFactory.java:108)
[a95ce0d211788da4b137109e03bbb3533782078d]   while locating org.apache.james.mailbox.cassandra.CassandraMailboxSessionMapperFactory
[a95ce0d211788da4b137109e03bbb3533782078d] 
[a95ce0d211788da4b137109e03bbb3533782078d] 9 errors
[a95ce0d211788da4b137109e03bbb3533782078d] 	at org.apache.james.mailbox.cassandra.mail.task.CassandraRecomputeCurrentQuotasServiceTest.setUp(CassandraRecomputeCurrentQuotasServiceTest.java:77)
```

**Comment by chibenwa** [comment]
```
[9d3f16becfe6b6f5e73beb5598abd9932a0d0d8a] [ERROR]   RabbitMQMailQueueTest$MailQueueSizeMetricsDisabled.constructorShouldNotRegisterGetQueueSizeGaugeWhenSizeMetricsDisabled » ParameterResolution

[9d3f16becfe6b6f5e73beb5598abd9932a0d0d8a] [ERROR] messagesShouldSurviveRabbitMQRestart  Time elapsed: 5.869 s  <<< ERROR!
[9d3f16becfe6b6f5e73beb5598abd9932a0d0d8a] org.junit.jupiter.api.extension.ParameterResolutionException: No ParameterResolver registered for parameter [org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration arg1] in method [void org.apache.james.queue.rabbitmq.RabbitMQMailQueueTest$MailQueueSizeMetricsEnabled.setup(org.apache.james.backends.cassandra.CassandraCluster,org.apache.james.backends.cassandra.init.configuration.CassandraConsistenciesConfiguration,org.apache.james.queue.api.MailQueueMetricExtension$MailQueueMetricTestSystem) throws java.lang.Exception].
```

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/configuration/CassandraConfiguration.java [hunk 6]
```diff
@@ -44,7 +46,10 @@
     public static final int DEFAULT_BLOB_PART_SIZE = 100 * 1024;
     public static final int DEFAULT_ATTACHMENT_V2_MIGRATION_READ_TIMEOUT = toIntExact(TimeUnit.HOURS.toMillis(1));
     public static final int DEFAULT_MESSAGE_ATTACHMENT_ID_MIGRATION_READ_TIMEOUT = toIntExact(TimeUnit.HOURS.toMillis(1));
-
+    public static final String DEFAULT_CONSISTENCY_LEVEL_REGULAR = "QUORUM";
```
Comment (mbaechler) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/configuration/CassandraConfiguration.java:49]:
```suggestion
    public static final String DEFAULT_CONSISTENCY_LEVEL_QUORUM = "QUORUM";
```

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/configuration/CassandraConfiguration.java [hunk 7]
```diff
@@ -57,6 +62,8 @@
     private static final String BLOB_PART_SIZE = "mailbox.blob.part.size";
     private static final String ATTACHMENT_V2_MIGRATION_READ_TIMEOUT = "attachment.v2.migration.read.timeout";
     private static final String MESSAGE_ATTACHMENTID_READ_TIMEOUT = "message.attachmentids.read.timeout";
+    private static final String CONSISTENCY_LEVEL_REGULAR = "cassandra.consistency_level.regular";
```
Comment (mbaechler) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/configuration/CassandraConfiguration.java:65]:
```suggestion
    private static final String CONSISTENCY_LEVEL_QUORUM = "cassandra.consistency_level.quorum";
```

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/configuration/CassandraConfiguration.java [hunk 8]
```diff
@@ -205,7 +238,9 @@ public CassandraConfiguration build() {
                 fetchNextPageInAdvanceRow.orElse(DEFAULT_FETCH_NEXT_PAGE_ADVANCE_IN_ROW),
                 blobPartSize.orElse(DEFAULT_BLOB_PART_SIZE),
                 attachmentV2MigrationReadTimeout.orElse(DEFAULT_ATTACHMENT_V2_MIGRATION_READ_TIMEOUT),
-                messageAttachmentIdsReadTimeout.orElse(DEFAULT_MESSAGE_ATTACHMENT_ID_MIGRATION_READ_TIMEOUT));
+                messageAttachmentIdsReadTimeout.orElse(DEFAULT_MESSAGE_ATTACHMENT_ID_MIGRATION_READ_TIMEOUT),
+                consistencyLevelRegular.orElse(DEFAULT_CONSISTENCY_LEVEL_REGULAR),
```
Comment (mbaechler) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/configuration/CassandraConfiguration.java:242]: we have to warn if DEFAULT_CONSISTENCY_LEVEL_QUORUM and DEFAULT_CONSISTENCY_LEVEL_LIGHTWEIGHT_TRANSACTION is a weird combination

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/configuration/CassandraConfiguration.java (hunk 6 above)
Comment (blackheaven) [reply backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/configuration/CassandraConfiguration.java:49]: I think it is mixing the aim and the value.

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/configuration/CassandraConfiguration.java (hunk 8 above)
Comment (blackheaven) [reply backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/configuration/CassandraConfiguration.java:242]: define weird

Comment (mbaechler) [reply backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/configuration/CassandraConfiguration.java:242]: LOCAL_QUORUM + SERIAL or LOCAL_SERIAL + QUORUM or EACH_QUORUM + LOCAL_SERIAL I guess

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/configuration/CassandraConfiguration.java (hunk 6 above)
Comment (mbaechler) [reply backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/configuration/CassandraConfiguration.java:49]: these are 3 quorum types, QUORUM is a specific QUORUM type

**Comment by chibenwa** [comment]
```
[d8a02a89e52c3f2dc752fb3fc9501fef2d6f91c1] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:testCompile (default-testCompile) on project apache-james-mailbox-cassandra: Compilation failure
[d8a02a89e52c3f2dc752fb3fc9501fef2d6f91c1] [ERROR] /james-project/mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraACLMapperTest.java:[250,15] exception org.apache.james.mailbox.exception.MailboxException is never thrown in body of corresponding try statement
[d8a02a89e52c3f2dc752fb3fc9501fef2d6f91c1] [ERROR] -> [Help 1]
```

**Comment by blackheaven** [comment]
weird issue

test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**